test/blackbox/run_blackbox_tests/test_cases_autoconnect.c

```diff
@@ -0,0 +1,156 @@
+/*
+    test_cases_blacklist.c -- Execution of specific meshlink black box test cases
+    Copyright (C) 2018  Guus Sliepen <guus@meshlink.io>
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+/* Modify this to change the logging level of Meshlink */
+#define TEST_MESHLINK_LOG_LEVEL MESHLINK_DEBUG
+
+#include "execute_tests.h"
+#include "test_cases_autoconnect.h"
+#include <pthread.h>
+#include "../common/test_step.h"
+#include "../common/common_handlers.h"
+#include "../../utils.h"
+#include <stdlib.h>
+#include <stdarg.h>
+#include <setjmp.h>
+#include <cmocka.h>
+#include <assert.h>
+#include <string.h>
+
+static void test_case_autoconnect(void **state);
+static bool test_steps_mesh_autoconnect(void);
+static meshlink_handle_t *mesh1, *mesh2;
+
+/* State structure for meshlink_blacklist Test Case #1 */
+static black_box_state_t test_mesh_autoconnect_state = {
+	.test_case_name = "test_case_mesh_autoconnect",
+};
+struct sync_flag test_autoconnect_m1n1_reachable = {.mutex  = PTHREAD_MUTEX_INITIALIZER, .cond = PTHREAD_COND_INITIALIZER, .flag = false};
+struct sync_flag test_autoconnect_blacklisted = {.mutex  = PTHREAD_MUTEX_INITIALIZER, .cond = PTHREAD_COND_INITIALIZER, .flag = false};
+struct sync_flag test_autoconnect_successful = {.mutex  = PTHREAD_MUTEX_INITIALIZER, .cond = PTHREAD_COND_INITIALIZER, .flag = false};
+
+/* Execute meshlink_blacklist Test Case # 1*/
+void test_case_autoconnect(void **state) {
+	execute_test(test_steps_mesh_autoconnect, state);
+}
+
+void callback_logger(meshlink_handle_t *mesh, meshlink_log_level_t level,
+                     const char *text) {
+	(void)level;
+
+	fprintf(stderr, "%s: {%s}\n", mesh->name, text);
+
+	if((check_sync_flag(&test_autoconnect_blacklisted) == true) && (strcmp("m1n2", mesh->name) == 0) && (strcmp("* could not find node for initial connect", text) == 0)) {
+		fprintf(stderr, "Test case successful\n");
+	  	set_sync_flag(&test_autoconnect_successful, true);
+	} else if((check_sync_flag(&test_autoconnect_blacklisted) == true) && (strcmp("m1n2", mesh->name) == 0)) {
+		assert(strcmp(text, "Autoconnect trying to connect to m1n1") != 0);
+	}
+
+}
+
+static void receive(meshlink_handle_t *mesh, meshlink_node_t *src, const void *data, size_t len) {
+	(void)mesh;
+	const char *msg = data;
```

Comment:
Instead of `const char *msg = data`, I think you should put `(void)src; (void)data` here. These are two unused variables. Otherwise, with full warnings enabled, the compiler will complain about them.